src/test/java/com/redhat/rcm/offliner/ftest/SinglePOMDepsDownloadFTest.java

```diff
@@ -92,11 +87,21 @@ public void run()
         // run `new Main(opts).run()` and return the Main instance so we can query it for errors, etc.
         Main finishedMain = run( opts );
 
-        assertThat( "Wrong number of downloads logged. Should have been 2 (declared jar + its corresponding POM).", finishedMain.getDownloaded(), equalTo( 2 ) );
+        assertThat( "Wrong number of downloads logged. Should have been 2 (declared jar + its corresponding POM).",
+                    finishedMain.getDownloaded(), equalTo( 2 ) );
         assertThat( "Errors should be empty!", finishedMain.getErrors().isEmpty(), equalTo( true ) );
 
         File downloaded = new File( downloads, path );
         assertThat( "File: " + path + " doesn't seem to have been downloaded!", downloaded.exists(), equalTo( true ) );
-        assertThat( "Downloaded file: " + path + " contains the wrong content!", FileUtils.readFileToByteArray( downloaded ), equalTo( content ) );
+        assertThat( "Downloaded file: " + path + " contains the wrong content!",
+                    FileUtils.readFileToByteArray( downloaded ), equalTo( content ) );
+
+        //re-run to test the function of avoiding re-downloading existing files
+        Main aviodedMain = run( opts );
```

Comment:
And again here, if we can copy the test class, rename to something descriptive about avoided downloads, and add this logic to that new class, it would be better.